File: server/src/main/java/org/cleanas2/service/net/util/HeaderBuilder.java
package org.cleanas2.service.net.util;

import org.apache.http.Header;
import org.apache.http.message.BasicHeader;

import java.util.ArrayList;

/**
 * @author Andrew Backer {@literal awbacker@gmail.com / andrew.backer@powere2e.com}
 */
public class HeaderBuilder {
    private final ArrayList<BasicHeader> headers = new ArrayList<>(10);

    public void add(String name, String value) {
        if (value != null) {
            headers.add(new BasicHeader(name, value));
        }
    }

    public Header[] toArray() {
        return headers.toArray(new Header[headers.size()]);
    }
}

File: server/src/main/java/org/cleanas2/service/net/util/HeaderBuilder.kt
package org.cleanas2.service.net.util

import org.apache.http.Header
import org.apache.http.message.BasicHeader

import java.util.ArrayList

/**
 * @author Andrew Backer awbacker@gmail.com / andrew.backer@powere2e.com
 */
class HeaderBuilder {
    private val headers = ArrayList<BasicHeader>(10)

    fun add(name: String, value: String?) {
        if (value != null) {
            headers.add(BasicHeader(name, value))
        }
    }

    fun toArray(): Array<Header> {
        return headers.toTypedArray()
    }
}
